benchdnn: introduce --summary knob for better usability (#2544)
oneapi-src/oneDNN: tests/benchdnn/doc/knob_summary.md

# Summary information

## Usage
```
    --summary=[no-]SETTING1[+[no-]SETTING2...]
```

The `--summary` knob is a global state of benchdnn and provides the summary
statistics at the end of the run. Different options are separated with `+`

> delimiter. To negate the effect of the option, "no-" prefix must be used in

Reviewer: ```suggestion
delimiter. To negate the effect of the option, use the "no-" prefix in
```

front of the option value.

> In case of same setting specified multiple times, only the latter value will be

Reviewer: ```suggestion
If the same setting is specified multiple times, only the latter value will be
```

considered.

## Failed cases summary

### Introduction
The number of test cases in a batch file can be big enough that running the
batch may result in the beginning of the list gets lost due to the short session
screen buffer. Even if the buffer could fit all the test cases in it, to find
the specific failed or unimplemented cases in the middle of the output usually
requires additional steps such as copying the whole screen buffer to a memory

> buffer and pasting it to some editor and using the search functionality.

Reviewer: ```suggestion
buffer, collecting the data in a file, and using the search functionality to scan through the file.
```

Author: Thank you for the edits, incorporated.

> `failures` knob comes handy to improve usability for such scenarios.

Reviewer: ```suggestion
The `failures` knob can improve usability in such scenarios.
```
